File: tests/testConfig.js
export const TEST_CONFIG = {
    // API Keys (should be loaded from environment variables)
    apiKeys: {
        openai: process.env.OPENAI_API_KEY,
        elevenlabs: process.env.ELEVENLABS_API_KEY,
        deepl: process.env.DEEPL_API_KEY,
        azure: process.env.AZURE_API_KEY,
        google: process.env.GOOGLE_API_KEY
    },
    
    // Test data paths
    fixtures: {
        audio: './tests/fixtures/test_audio.wav',
        text: './tests/fixtures/test_text.txt'
    },
    
    // Performance thresholds (in milliseconds)
    performance: {
        sttLatency: 1000, // 1 second
        translationLatency: 500, // 0.5 seconds
        ttsLatency: 1500, // 1.5 seconds
        totalPipelineLatency: 4000 // 4 seconds
    },
    
    // Rate limiting
    rateLimits: {
        stt: 10, // requests per minute
        translation: 50, // requests per minute
        tts: 20 // requests per minute
    },
    
    // Supported languages
    languages: [
        'en', 'fr', 'es', 'de', 'it', 'nl', 'pt', 'ru', 'zh', 'ja'
    ],
    
    // Test environment settings
    environment: {
        timeout: 10000, // 10 seconds
        retries: 3,
        parallel: true
    }
};

File: tests/test_config.py
import os
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

class TestConfig:
    # API Keys
    API_KEYS = {
        'OPENAI_API_KEY': os.getenv('OPENAI_API_KEY'),
        'ELEVENLABS_API_KEY': os.getenv('ELEVENLABS_API_KEY'),
        'DEEPL_API_KEY': os.getenv('DEEPL_API_KEY'),
        'AZURE_API_KEY': os.getenv('AZURE_API_KEY'),
        'GOOGLE_API_KEY': os.getenv('GOOGLE_API_KEY')
    }
    
    # Test data paths
    TEST_AUDIO_PATH = 'tests/fixtures/test_audio.wav'
    TEST_TEXT_PATH = 'tests/fixtures/test_text.txt'
    
    # Performance thresholds
    STT_LATENCY_THRESHOLD = 1.0  # seconds
    TRANSLATION_LATENCY_THRESHOLD = 0.5  # seconds
    TTS_LATENCY_THRESHOLD = 1.5  # seconds
    TOTAL_PIPELINE_LATENCY_THRESHOLD = 4.0  # seconds
    
    # Rate limiting
    RATE_LIMITS = {
        'STT': 10,  # requests per minute
        'TRANSLATION': 50,  # requests per minute
        'TTS': 20  # requests per minute
    }
    
    # Test languages
    SUPPORTED_LANGUAGES = [
        'en', 'fr', 'es', 'de', 'it', 'nl', 'pt', 'ru', 'zh', 'ja'
    ]
